homeassistant/components/alarm_control_panel/arlo.py

```diff
@@ -0,0 +1,109 @@
+"""
+This component provides HA alarm_control_panel support for Arlo System.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/alarm_control_panel.arlo/
+"""
+import asyncio
+import logging
+import voluptuous as vol
+
+from homeassistant.components.arlo import (DATA_ARLO, CONF_ATTRIBUTION)
+from homeassistant.components.alarm_control_panel import (AlarmControlPanel,
+                                                          PLATFORM_SCHEMA)
+from homeassistant.const import (ATTR_ATTRIBUTION, STATE_ALARM_ARMED_AWAY,
+                                 STATE_ALARM_ARMED_HOME, STATE_ALARM_DISARMED)
+import homeassistant.helpers.config_validation as cv
+
+DEPENDENCIES = ['arlo']
+
+_LOGGER = logging.getLogger(__name__)
+
+CONF_HOME_MODE_NAME = 'home_mode_name'
+ICON = 'mdi:security'
+ARMED = "armed"
+DISARMED = "disarmed"
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Optional(CONF_HOME_MODE_NAME, default=ARMED): cv.string,
```

Comment:
Is this configurable on the Arlo device? Is that why this should be configurable here?

Reply:
Arlo base station only has two modes by default. Armed and Disarmed. But users can create custom modes on the arlo app. Since HA alarm panel component has the notion of 'home' mode, this configuration just let's them map one of their 'custom' modes as 'home' mode. If they don't have one, then home and away modes are the same (and the status will always say armed_away).

Sorry if i wasn't clear in the description. :)